src/ImageSharp/Formats/Png/PngDecoderCore.cs

```diff
@@ -241,29 +259,79 @@ public IImageInfo Identify(BufferedReadStream stream, CancellationToken cancella
                         {
                             case PngChunkType.Header:
                                 this.ReadHeaderChunk(pngMetadata, chunk.Data.GetSpan());
+                                this.isHeaderChunkReached = true;
                                 break;
                             case PngChunkType.Physical:
+                                if (this.colorMetadataOnly)
+                                {
+                                    this.SkipChunkDataAndCrc(chunk);
+                                    break;
+                                }
+
                                 this.ReadPhysicalChunk(metadata, chunk.Data.GetSpan());
                                 break;
                             case PngChunkType.Gamma:
+                                if (this.colorMetadataOnly)
+                                {
+                                    this.SkipChunkDataAndCrc(chunk);
+                                    break;
+                                }
+
                                 this.ReadGammaChunk(pngMetadata, chunk.Data.GetSpan());
                                 break;
                             case PngChunkType.Data:
                                 this.SkipChunkDataAndCrc(chunk);
```

Comment:
[Spec says](http://www.libpng.org/pub/png/spec/1.2/PNG-Chunks.html#C.tRNS):

> When present, the tRNS chunk must precede the first IDAT chunk, and must follow the PLTE chunk, if any.

It means that if we reach Data, we are free to quit the loop. Forwarding the file position beyond the data chunk may have IO overhead we can easily avoid here.